climada/engine/impact.py

```diff
@@ -911,9 +915,10 @@ def _build_exp_event(self, event_id):
         Parameters:
             event_id(int): id of the event
         """
+        ix = np.isin(self.event_id, event_id).nonzero()[0]
```

Comment:
After this, `ix` will be a 1-d array, right? Fortunately, it has only a single entry, but it's still confusing. Furthermore, I find it even more confusing to use `np.isin` with a scalar. Why don't you use the following?
```suggestion
        [[ix]] = (self.event_id == event_id).nonzero()
```

Reply:
Just a general interest question: is `ix` a standard abreviation for index in python coding? I usually use `idx`. Probably we should just use what is more comon.

Reply:
`argwhere` and `nonzero` are not equivalent, in general. They are equivalent in this particular case. However, I think that, in most use cases, you are interested in all indices of nonzero entries, most probably in a 1-d array. And that is what you get from `[ix] = (...).nonzero()` while it has to be the rather awkward `ix = np.argwhere(...)[:,0]`.

In my opinion, `argwhere` is quite useless and I never use it. But there might be relevant use cases somewhere out there. Still, I wouldn't tell anyone in a review to prefer `nonzero` over `argwhere` unless the result is wrong. But outside of a review and in general use, I would always recommend to stick to `nonzero` and only use `argwhere` if there is a palpable advantage.

Reply:
Fair enough. I always found it weird to use `nonzero` to test True/False values.

Reply:
I understand that you find the naming not very intuitive, but this has been around forever. It's also mentioned in the docs: "A common use for nonzero is to find the indices of an array, where a condition is True" (https://numpy.org/doc/stable/reference/generated/numpy.nonzero.html). Everybody who is working with NumPy for a while gets used to it. It's rather irritating NOT to use `nonzero` :smile:

Reply:
@tovogt I'll use your suggestion, it's a good improvement. with From my R days, it still feels weird to convert a boolean array to a set of indices because they're equivalent for subsetting.

Reply:
ack, committed chahan's suggestion instead 😩
there are like, four more commits now, sorry.